lesson_05/script.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+const ul = document.getElementById("js-list");
+const fragment = document.createDocumentFragment();
+const data = [
+  { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+];
+
+const promise = new Promise((resolve) => {
+  resolve(data);
+}).then((value) => {
+  for (let i = 0; i < value.length; i++) {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+    a.href = value[i].to;
+    a.text = value[i].text;
+    img.src = value[i].img
+    img.alt = value[i].alt;
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
+  }
+  ul.appendChild(fragment);
```

Comment:
NITS
I thought it would be easier to write a promise and how to do it separately. What do you think?
```suggestion
const promise = new Promise((resolve) => {
  resolve(data);
});
promise.then((value) => {
  for (let i = 0; i < value.length; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");
    a.href = value[i].to;
    a.text = value[i].text;
    img.src = value[i].img
    img.alt = value[i].alt;
    fragment
      .appendChild(li)
      .appendChild(a)
      .insertAdjacentElement("afterbegin", img);
  }
  ul.appendChild(fragment);
```

Reply:
ありがとうございました。こちらの方が読みやすく、勉強になります。[`0faba26`](https://github.com/Numabeharuka/Lesson-js/pull/6/commits/0faba26439b553640a4668b726194bc33453a41e)